skiplang/compiler/src/skipTypedAstExpand.sk

```diff
@@ -171,8 +171,7 @@ fun type__inter(
         Some(elt)
       }
     })
-    .filter(opt -> opt is Some _)
-    .map(opt -> (opt as Some _).value)
+    .filterNone()
```

Comment:
:-( This would be good if `filterNone` wasn't misnamed: it is confusing that `filter` keeps the elements that for which the predicate is true while `filterNone` is the opposite keeping the elements that are not `None`.

Reply:
I agree, be me renaming guest :D